service/waitlist/waitlist.go

```diff
@@ -126,6 +129,18 @@ func (s *service) DeleteItem(waitlistID, itemID []byte, reason string) error {
 
 // CreateItem creates a new item in a waitlist
 func (s *service) CreateItem(waitlistID []byte, item *models.Item) (*models.Item, error) {
+	currentItems, err := s.storage.ListItems(waitlistID)
+	if err != nil {
+		return nil, err
+	}
+
+	// check with the item with provided patientID already exists
+	for i := range currentItems {
+		if *currentItems[i].PatientID == *item.PatientID {
+			return nil, utils.NewError(utils.ErrConflict, "Waitlist already contains item with provided patientID")
+		}
+	}
+
 	return s.storage.AddItem(waitlistID, item)
 }
```

Comment:
This just shortens the window in which the race condition can occur. Should we move this check inside the bolt transaction in the storage?

Reply:
ok